File: CPython/剑指 Offer 04. 二维数组中的查找.py
class Solution:
    def findNumberIn2DArray(self, matrix: List[List[int]], target: int) -> bool:
    	i, j = len(matrix), 0
    	while  i > 0 && j < len(matrix[0]):
    		if matrix[i][j] < target: j+=1
    		elif matrix[i][j] > target : i-=1
    		else return True
    	return False

File: CPython/剑指 Offer 04. 二维数组中的查找.cpp
class Solution {
public:
    bool findNumberIn2DArray(vector<vector<int>>& matrix, int target) {
    	int i = matrix.size()-1;
    	int j = 0;
    	while (i >= 0 && j < matrix[0].size()){
    		if (matrix[i][j] > target) j--;
    		else if (matrix[i][j] < target) i++;
    		return true; 
    	}
    	return false;
    }
};